### PR TITLE
O(n) for comments :^)

### PR DESCRIPTION
**Pull request by TeoReu**

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -158,45 +158,62 @@ function deleteComment(comment) {
...
-
-  xhr.open('GET', '/data', true);        // Prepare the request
-  xhr.send(null);                                 // Send the request
+  xhr.open('GET', '/data', true);        
+  xhr.send(null);                               
 }
 
 // Refreshes the data each 2 seconds
-setInterval(getData, 1000*2);
+const milsec = 2;
```
Comment (chripell) [inline portfolio/src/main/webapp/script.js:173]: It is sec not milsec.

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -158,45 +158,62 @@ function deleteComment(comment) {
...
 
 // New comments(=comments) are added to the page, and old comments(=currentComments) are removed
 function refreshComments(comments){
-  for(var i = 0; i < currentComments.length; i++){
-    if(objNotInList(currentComments[i],comments)){
-      removeCommentElement(i);
-      currentComments.splice(i,1);
+  var i = comments.length - 1;
+  var j = currentComments.length - 1;
+  while(currentComments[0].timestamp >= comments[i].timestamp){
```
Comment (chripell) [inline portfolio/src/main/webapp/script.js:180]: What happens, for example, when the last comment is deleted? I am a bit worried about the termination condition for the loop.

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -158,45 +158,62 @@ function deleteComment(comment) {
...
+  //newComment = createCommentElement(comments[j]);
+  // const commentsListElement = document.getElementById('comments-list');
+  //commentsListElement.insertBefore(newComment,commentsListElement.firstChild);      
+  //removeCommentElement(i);
+  //currentComments.splice(i,1);
   currentComments = comments;
 }
 
 // Check if a comment is in a list of comments, by comparing timestamps
 function objNotInList(comment, comments){
```
Comment (chripell) [inline portfolio/src/main/webapp/script.js:215]: Is this function still needed now?

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (TeoReu) [reply portfolio/src/main/webapp/script.js:180]: Well if we have: (a3> a2>a1) timestamps
comments = { a2 , a1}
Cc = {a3, a2, a1}
for i =  1 and j = 2 (we are going to have a1=a1)
for i = 0 and j =1 (a2=a2)
for i = 0 and j = 0 (we remove the visual element, but also remove a3 from the list)
for i = 0 and j = 0 ( we have a2 = a2 -> it enters on the branch with the break of the if)
It is a bit shady, but my argument is that the algorithm leads to case i = j = 0. If the comments are different removeElement() and if they are the same than break;

Comment (TeoReu) [reply portfolio/src/main/webapp/script.js:180]: What do you think?

Comment (TeoReu) [reply portfolio/src/main/webapp/script.js:180]: I will try to think of something better tho (less not-clear, bcs it's not very clear how it works for me neither)

**Comment by chripell** [comment]
As discussed, ping me when you are ready for another review.

**Pull request merged**